waku/factory/node_factory.nim

```diff
@@ -144,14 +145,23 @@ proc setupProtocols(
 
     peerExchangeHandler = some(handlePeerExchange)
 
-  let shards =
-    conf.contentTopics.mapIt(node.wakuSharding.getShard(it).expect("Valid Shard"))
-  debug "Shards created from content topics",
-    contentTopics = conf.contentTopics, shards = shards
+  var pubsubTopics: seq[string] = conf.pubsubTopics
+  for shard in conf.shards:
+    pubsubTopics.add($NsPubsubTopic.staticSharding(conf.clusterId, shard))
+
+  if conf.shards.len == 0:
+    # If shards are not configured but content topics are, then assume autosharding is expected
```

Comment:
I am guessing this is incorrect as you'd need to have shards configured (ie 0 to 7 ) to apply an auto sharding logic as the result of the autosharding logic depends on the shard range available.